lm-api/lm_api/security.py

```diff
@@ -46,14 +49,38 @@ def extract_organization(self) -> Self:
                 }
             }
         }
+        
+        or:
+
+        # New json structure
+        {
+            ...,
+            "organization": {
+                "orgname": {
+                    "id": "adf99e01-5cd5-41ac-a1af-191381ad7780",
+                    ...
+                }
+            }
+        }
+
         """
         if self.organization is None:
             return typing.cast(Self, self)
         elif len(self.organization) != 1 and isinstance(self.organization, dict):
             raise ValueError(
                 f"Organization payload did not include exactly one value: {getattr(self, 'organization')}"
             )
-        setattr(self, "organization_id", next(iter(self.organization)))
+
+        if isinstance(self.organization, dict):
+            org_field = next(iter(self.organization))
+            if re.fullmatch(r'^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$', org_field):
```

Comment:
This regex can be simplified to use `\d` instead of `0-9` and group the 3 instances of 4-characters groups:

```suggestion
            if re.fullmatch(r'^[\da-fA-F]{8}(?:-[\da-fA-F]{4}){3}-[\da-fA-F]{12}$', org_field):
```
Also, instead of having this regex being explicitly used here, it could be moved somewhere else. Maybe to the `constants.py` module?